librz/util/subprocess.c

```diff
@@ -41,6 +41,7 @@ struct rz_subprocess_t {
 	HANDLE stderr_read;
 	HANDLE proc;
 	int ret;
+	RzThreadSemaphore *ret_sem;
```

Comment:
I'm not so sure this is the correct way. I would expect this to be not thread safe and have a semaphore/lock where this is tossed into a threaded code.
Also this change should be done in rz_sys_fork etc.. Where we use a mutex but it should be a semaphore

Reply:
> I would expect this to be not thread safe and have a semaphore/lock where this is tossed into a threaded code.

With d2ad89da523df4a327b12367273e8658a4227f6e, there is a semaphore whether or not the calling code is multithreaded.

> Also this change should be done in rz_sys_fork etc.. Where we use a mutex but it should be a semaphore

The mutex in rz_sys_fork seems to be for a purpose unrelated to the data race in the [description](https://github.com/rizinorg/rizin/pull/4545#issue-2342038472). There might be a race there too though.